### PR TITLE
fix webapi#json#call

### PR DESCRIPTION
**Pull request by liushapku**
#32 had a quick fix by adding `varargs` to capture the optional `id`

### CODE REVIEW

File: autoload/webapi/jsonrpc.vim [hunk 1]
```diff
@@ -8,13 +8,24 @@
 let s:save_cpo = &cpo
 set cpo&vim
 
-function! webapi#jsonrpc#call(uri, func, args) abort
-  let data = webapi#json#encode({
+function! webapi#jsonrpc#call(uri, func, args, ...) abort
+  let request = {
```
Comment (mattn) [inline autoload/webapi/jsonrpc.vim]: let opts = a:1

File: autoload/webapi/jsonrpc.vim [hunk 2]
```diff
@@ -8,13 +8,24 @@
...
   \ 'jsonrpc': '2.0',
   \ 'method':  a:func,
   \ 'params':  a:args,
```
Comment (mattn) [inline autoload/webapi/jsonrpc.vim]: 'id': has_key(opts, 'id') ? opts['id'] : 0,

**Comment by mattn** [comment]
For the futher, I want to pass extra argument as dictionary.

**Comment by mattn** [comment]
And if you expect the number for extra argument,

```
let opts = a:1
let id = type(opts) == type({}) ? (has_key(opts) ? opts['id'] : 0) : opts
```

File: autoload/webapi/jsonrpc.vim (hunk 2 above)
Comment (liushapku) [reply autoload/webapi/jsonrpc.vim]: then 'id' will default to 0, we will never be able to fire notification.

**Comment by liushapku** [comment]
Personally, I feel it is not necessary to accept additional options. Adding that complicate things. 

You can decide. Adding that is not harmful. We can use it to config what is null, etc.

**Pull request closed without merging**